### PR TITLE
add ga4 mapping for va-select and va-radio

### PR DESCRIPTION
**Pull request by it-harrison**
This PR adds GA4 mapping for `va-select` and `va-radio` DST web components

## Summary
updates `component-library-analytics-setup.js`

## Testing done
local

## Screenshots
va-select GA4 event:
Screenshot 2023-07-10 at 12 31 09 PM

va-radio GA4 event:
Screenshot 2023-07-10 at 3 37 41 PM

## What areas of the site does it impact?

any page that uses `va-select` or `va-radio` web components

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/platform/site-wide/component-library-analytics-setup.js [hunk 1]
```diff
@@ -464,6 +464,18 @@ const analyticsEvents = {
...
+        event: 'interaction',
+        component_name: 'va-radio',
+        custom_string_1: 'component_library',
+      },
+      /* Component to GA4 parameters */
+      mapping: {
+        'radio-button-label': 'heading_1',
+        'radio-button-optionLabel': 'value',
+        'radio-button-required': 'required',
+        version: 'component_library',
```
Comment (jamigibbs) [inline src/platform/site-wide/component-library-analytics-setup.js:477]: Even though we aren't really passing a component_version, this mapping has been getting set to `version: 'component_version',` for the others.

**Comment by it-harrison** [comment]
> @it-harrison Were you able to test this locally? A screenshot added to the description would be helpful to visualize what is being pushed to the datalayer. We have instructions on how to do that here (How to Test GA and GA4 Locally): https://vfs.atlassian.net/wiki/spaces/DST/pages/2544173086/Adding+GA4+details+for+Google+Analytics

thanks will check it out

File: src/platform/site-wide/component-library-analytics-setup.js [hunk 2]
```diff
@@ -464,6 +464,18 @@ const analyticsEvents = {
...
+      ga4: {
+        event: 'interaction',
+        component_name: 'va-radio',
+        custom_string_1: 'component_library',
+      },
+      /* Component to GA4 parameters */
+      mapping: {
+        'radio-button-label': 'custom_string_2',
+        'radio-button-optionLabel': 'value',
+        'radio-button-required': 'required',
```
Comment (jamigibbs) [inline src/platform/site-wide/component-library-analytics-setup.js:476]: I don't think these are mapping correctly. Should the prefix just be `radio-label`, `radio-optionLabel`, `radio-required`?

```
'radio-button-label': 'custom_string_2',
'radio-button-optionLabel': 'value',
'radio-button-required': 'required',
```

Screenshot 2023-07-10 at 2 48 37 PM

I would expect to see them in the console like:

```
custom_string_2: "something",
value: "something",
required: false
```

Comment (it-harrison) [reply src/platform/site-wide/component-library-analytics-setup.js]: good catch - fixed!

**Review by jamigibbs — APPROVED** [review]

**Review by CBonade — APPROVED** [review]

**Pull request merged**